# README for Graded Assignment 2

Instructions copied from assignment ([source](https://mcic-osu.github.io/pracs-sp21/w06_GA_scripts.html))

> This assignment will work with 6 FASTQ files with sequences from the V4 region of 16S rRNA, generated in a metabarcoding experiment.

The FASTQ files come in pairs: for every sample, there is a FASTQ file with forward reads (or “read 1” reads) that contains _R1_ in its file name, and a FASTQ file with corresponding reverse reads (or “read 2” reads) that contains _R2_ in its file name. So, our 6 FASTQ files consist of 3 pairs of R1-R2 files for 3 biological samples.

The sequences were generated by first amplifying environmental samples with a pair of universal 16S primers, and these primer sequences are expected to be present in the FASTQ sequences. You will remove these primer sequences with the program Cutadapt, and there are two things to be aware of:

A primer can also be present in the FASTQ sequence as its reverse complement, so we will search for reverse complements too.
The primers contain a few variable sites for which ambiguity codes are used. For instance, an R means that the site can be either an A or a G, and an N means that the site can be any of the four bases. See here for a complete overview of these ambiguity codes.
These are the primer sequences 1 :

Forward primer (“515F”): GAGTGYCAGCMGCCGCGGTAA.
Reverse primer (“806R”): ACGGACTACNVGGGTWTCTAAT.
General instructions

For each numbered step below, you should create at least one Git commit.

All files should be added to your repository, unless mentioned otherwise.


